certbot-apache/certbot_apache/configurator.py

```diff
@@ -1377,7 +1478,18 @@ def enhance(self, domain, enhancement, options=None):
             raise errors.PluginError(
                 "Unsupported enhancement: {0}".format(enhancement))
         try:
-            func(self.choose_vhost(domain), options)
+            if self.wildcard_domain(domain):
+                # Handle virtualhosts configured for a wildcard domain
+                if domain in self.wildcard_vhosts.keys():
```

Comment:
can this if (`if domain in self.wildcard_vhosts:`) go inside `choose_vhosts_wildcard`?

Reply:
oh yeah, if both this and the other thing go into the function, that cleans everything else up a bunch!


these are my thoughts as I do the nginx one, it may be clearer to look at the PR I put up for that